Here strings are not supported by busybox ash as of version 1.31.0.
Substitute with echo and piping.

---
--- setup-2.13.6/lang.sh.orig	2020-06-22 16:50:05.945742987 +0300
+++ setup-2.13.6/lang.sh	2020-06-22 16:50:29.398989790 +0300
@@ -59,7 +59,7 @@
 
 # The ${LANG} manipulation is necessary only in virtual terminal (a.k.a. console - /dev/tty*):
 if [ -n "${LANG}" ] && [ "${TERM}" = 'linux' ] && /usr/bin/tty | /usr/bin/grep --quiet -e '/dev/tty'; then
-    if /usr/bin/grep --quiet -E -i -e '^.+\.utf-?8$' <<< "${LANG}"; then
+    if echo "${LANG}" | /usr/bin/grep --quiet -E -i -e '^.+\.utf-?8$'; then
         case ${LANG} in
             ja*)    LANG=en_US.UTF-8 ;;
             ko*)    LANG=en_US.UTF-8 ;;
